bindings/matrix-sdk-ffi/src/room.rs

```diff
@@ -902,6 +902,20 @@ impl Room {
             Ok(Arc::new(RoomMessageEventContent::new(msgtype)))
         })
     }
+
+    pub async fn can_user_redact(&self, user_id: String) -> Result<bool, ClientError> {
+        let room = match &self.inner {
+            SdkRoom::Joined(j) => j.clone(),
+            _ => return Err(anyhow!("Can't check permissions for non joined rooms").into()),
+        };
+
+        let user_id = UserId::parse(&user_id)?;
+        Ok(room.can_user_redact(&user_id).await?)
+    }
```

Comment:
I tried exposing this directly as uniFFI async, however I get this error when using this API, so I am probably doing it wrong:
thread '<unnamed>' panicked at 'there is no reactor running, must be called from the context of a Tokio 1.x runtime', /Users/mauroromito/.cargo/registry/src/index.crates.io-6f17d22bba15001f/deadpool-runtime-0.1.2/src/lib.rs:74:29
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread '<unnamed>' panicked at 'there is no reactor running, must be called from the context of a Tokio 1.x runtime', /Users/mauroromito/.cargo/registry/src/index.crates.io-6f17d22bba15001f/deadpool-runtime-0.1.2/src/lib.rs:100:22


Is it worth to exposing it directly (and if it is what is the best way) as async or should I refrain to do from now?

Reply:
You need to change `#[uniffi::export]` to `#[uniffi::export(async_runtime = "tokio")]` on line 71.

Reply:
You shouldn't have to split up impl blocks. If you get errors locally, try rebasing.

Reply:
It won't affect non-`async` function and we never want `async` function to run without the tokio runtime context. Please join all the impl blocks again.